tests/paragraph.spec.ts

```diff
@@ -90,6 +91,33 @@ test('backspace and arrow on title', async ({ page }) => {
   await assertTitle(page, 'hll');
 });
 
+test('backspace on line start of the first block', async ({ page }) => {
```

Comment:
This is a test case that is expected for this edge case. You can use the [playwright VSCode plugin](https://marketplace.visualstudio.com/items?itemName=ms-playwright.playwright) to debug it visually. Quality-related takeaways here:

* To ensure the state is stable, we require that the editor state should be the same after one undo and one redo, just like the [principle](https://www.figma.com/blog/how-figmas-multiplayer-technology-works/#:~:text=if%20you%20undo%20a%20lot%2C%20copy%20something%2C%20and%20redo%20back%20to%20the%20present%20(a%20common%20operation)%2C%20the%20document%20should%20not%20change.) shared by Figma.
* If no paragraph remains on the page, there will be a cursor blind zone. So in the asserter, we expect `['\n']` (one empty paragraph or list block) instead of `[]` (no content block) here.